### PR TITLE
C# utlity classes for export. (Bottom-up)

### PR DESCRIPTION
**Pull request by mr-phazer**
Add; 
- `PackedMeshBuilder `updated to be more S.O.L.I.D, it is now "open closed". Its interface `IPackedMeshBuilder `has two implementations, `IndexedPackedMeshBuilder`, and `WeightedIndexedPackedMeshBuilder`.  the scene builder uses
`PackedMeshBuilderFactory.GetBuilder(AnimationFile skeletonFile)` to obtain the riggedd of non-rigged version as needed.

- `VertexWeightCreator` is used  to copy the 1-4 weights in an RMV2 vertex to 1 un-nested array, 

- Added  `Animations.GetIdFromBoneName(string name)` for save lookups (not out of range),  AnimationFile already has `string GetBoneNameFromId(int id)

- 2-3 minor helper classes

### CODE REVIEW

File: CommonControls/FileTypes/Animation/AnimationFile.cs [hunk 1]
```diff
@@ -181,10 +181,18 @@ public int GetIdFromBoneName(string name)
...
                 .FirstOrDefault();
-                            
+
             return (boneInfo == null) ? -1 : boneInfo.Id;
         }
 
+        /// <summary>
+        /// Safe method for looking up bones, as opposed to using the bone index directly
+        /// </summary>
+        public string GetBoneNameFromIndex(int boneIndex)
```
Comment (donkeyProgramming) [inline CommonControls/FileTypes/Animation/AnimationFile.cs:191]: this should not be added here. 
Feel free to add it as an extention method within the assetManagment project or as a helper function where used

**Review by donkeyProgramming — CHANGES_REQUESTED** [review]
Unless absolutly required, dont modify files outside of the assetmanagementproject

File: CommonControls/FileTypes/Animation/AnimationFile.cs (hunk 1 above)
Comment (mr-phazer) [reply CommonControls/FileTypes/Animation/AnimationFile.cs]: Yeah, oops,, I get it :)
I will remove both
`string GetBoneNameFromIndex(int boneIndex)`
`public int GetIdFromBoneName(string name)` 
from` AnimationFIle`
to `class AsseetMagement.AnimationProcessing.SkeletonHelper`

**Comment by mr-phazer** [comment]
Changes done:
- Moved:
  -`string GetBoneNameFromIndex(int boneIndex)`
  -`int GetIdFromBoneName(string name)`
from` class AnimationFIle`
to class `AsseetMagement.AnimationProcessing.SkeletonHelper`
- Changed reference to the methods in `AsseetMagement` to the new methods in `SkeletonHelper`

**Pull request merged**